warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RootNegationCheckVisitor.java

```diff
@@ -46,8 +46,7 @@ private boolean hasLeafWithoutNegation(JexlNode node) {
             
             if (ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass)) {
                 // no need to inspect this path anymore
-                continue;
```

Comment:
Why are we keeping the preceding if statement around?

Reply:
What I'm saying is that instead of keeping an empty if statement, why not just do something like this:

```
if (!(ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass)))  {
    if (child == null || child.jjtGetNumChildren() == 0) {
        // reached a leaf node without ever encountering a negation
        foundLeaf = true;
    } else {
        // neither a leaf nor negation, continue looking
        foundLeaf = hasLeafWithoutNegation(child);
    }
}
```

Reply:
This could be simplified further as:
```
if (!(ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass)))  {
    foundLeaf = (child == null || child.jjtGetNumChildren() == 0) || hasLeafWithoutNegation(child);
}
```

Reply:
Oh that's much nicer.

Reply:
Basically, instead of checking for what we don't want, and doing nothing, let's just check for the condition we do want, and do something.